### PR TITLE
Add support for Django 4.1

### PR DESCRIPTION
**Pull request by fizyk**
* Strip IDENTITY related suffixes auto fields are getting on Django 4.1.
      These suffixes are upsetting redshift.

Subject: Django 4.1 support

### Feature or Bugfix

- Feature

### Purpose
- Add support for Django 4.1

### Detail
- Clearing DatabaseWrapper.data_types_suffix which, on DJango 4.1, are responsible for adding "GENERATED BY DEFAULT AS IDENTITY" to auto fields, and which are upsetting Redshift. Actually makes impossible to create new table or django_migration table.

### Relates
- #106

### CODE REVIEW

**Comment by fizyk** [comment]
pre-commit is... failing

**Comment by fizyk** [comment]
Updated documentation and setup.cfg as well

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -11,10 +11,12 @@ jobs:
       max-parallel: 5
       matrix:
         python-version: ['3.7', '3.8', '3.9', '3.10']
-        django-version: ['3.2', '4.0']
+        django-version: ['3.2', '4.0', '4.1']
         exclude:
           - django-version: '4.0'
             python-version: '3.7'
+          - django-version: '4.1'
+            python-version: '3.7'
```
Comment (shimizukawa) [inline .github/workflows/test.yml:19]: memo: django 4.1 supports python 3.8 or later 
https://docs.djangoproject.com/en/4.1/releases/4.1/

File: tox.ini [hunk 2]
```diff
@@ -17,6 +17,7 @@ python =
 DJANGO =
     3.2: dj32
     4.0: dj40
+    4.1: dj41
```
Comment (shimizukawa) [inline tox.ini:20]: We also need dj41 in `envlist` (line 2) and `deps` (line 32)

Comment (shimizukawa) [reply tox.ini:20]: Done.

**Comment by shimizukawa** [comment]
> IDENTITY related suffixes auto fields are getting on Django 4.1

The suffix related to IDENTITY appeared because Django 4.1 now creates AutoField with `integer GENERATED BY DEFAULT AS IDENTITY` available since postgres 10 instead of `serial`.
https://github.com/django/django/blob/bc3b8f152452ba0e41f28baa93c0bf8f39cddb09/django/db/backends/postgresql/schema.py#L44-L47

Therefore, the current PR will create tables with `integer` instead of `serial`, which is a different behavior than in django-4.0 and earlier.
I've been investigating for a couple of hours and haven't found a good fix yet. Please let me know if there is a way to fix it.

**Comment by fizyk** [comment]
Hmmm... I thought the serial stayed since the tests checked the resulting SQL and without the suffixes it was the same as on earlier Django versions.

**Comment by fizyk** [comment]
Where should I look? The AutoField data_type was already overriden by redshift-engine to `integer identity(1, 1)` so i guess somewhere is a part that also needs to be changed for postgresql to either vendor, or ditch the serial in favour of a modern `GENERATED BY DEFAULT AS IDENTITY` 🤔

**Comment by fizyk** [comment]
@shimizukawa the way I see it is that the tests running directly on Postgres will become increasingly hard to maintain with each PostgreSQL/Django release (and redshift evolves too).

Maybe the best approach would be to fork the whole code from Django 4.0.x instead of inheriting from the Postgres engine from whatever Django is there underneath. Plus the Postgres implementation and redshift will become increasingly distinct 🤔

I tried to pinpoint what has changed between 4.0.x and 4.1.x but all I got was additional IntegerField generated with primary_key set to True. (copied `get_field_type` from 4.0.x)

**Comment by shimizukawa** [comment]
Thank you @fizyk,
I apologize for the long delay. I was able to complete the implementation of django-4.2 support based on your proposal to vend django code base.
I'm closing this PR now that Django-4.2 support is complete.
Thank you for your great contribution.

**Pull request closed without merging**